app/models/service_orchestration.rb

```diff
@@ -35,6 +35,23 @@ def deploy_orchestration_stack
     save_create_options
   end
 
+  def deploy_orchestration_stack_queue
+    userid ||= User.current_user.try(:userid)
+    task_opts = {
+      :action => "deploying Orchestration Stack for user #{userid}",
+      :userid => userid
+    }
+    queue_opts = {
+      :class_name  => self.class.name,
+      :method_name => 'deploy_orchestration_stack',
+      :instance_id => id,
+      :role        => 'ems_operations',
+      :zone        => orchestration_manager.my_zone,
+      :args        => []
+    }
+    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  end
+
   def update_orchestration_stack
```

Comment:
Similarly we should have a queue version for update too. It can be either included here or with a separate PR.